src/Tools/LeanCode.CodeAnalysis/CodeActions/FixCQRSHandlerNamespaceCodeAction.cs

```diff
@@ -88,40 +88,43 @@ CancellationToken cancellationToken
 
-            var currDirectoryIdx = currentDirectory.LastIndexOf(projectName, StringComparison.InvariantCulture);
-            var expectedNamespaceIdx = expectedNamespace.StartsWith(projectName, StringComparison.InvariantCulture)
-                ? 0
-                : -1;
-
-            if (currDirectoryIdx != -1 && expectedNamespaceIdx != -1)
+            if (currentDirectory != null)
             {
-                currDirectoryIdx = currDirectoryIdx + projectName.Length;
-                expectedNamespaceIdx = expectedNamespaceIdx + projectName.Length;
+                var currDirectoryIdx = currentDirectory.LastIndexOf(projectName, StringComparison.InvariantCulture);
+                var expectedNamespaceIdx = expectedNamespace.StartsWith(projectName, StringComparison.InvariantCulture)
+                    ? 0
+                    : -1;
 
-                // Path prefix with project name.
-                var pathPrefix = currentDirectory[..currDirectoryIdx];
-                // Namespace without project name prefix.
-                var namespaceSuffix = expectedNamespace[expectedNamespaceIdx..];
+                if (currDirectoryIdx != -1 && expectedNamespaceIdx != -1)
+                {
+                    currDirectoryIdx = currDirectoryIdx + projectName.Length;
+                    expectedNamespaceIdx += projectName.Length;
 
-                // `Path.Combine` will return the second argument if it begins with a separation character.
-                var pathSuffix = namespaceSuffix
-                    .Replace('.', Path.DirectorySeparatorChar)
-                    .Trim(Path.DirectorySeparatorChar);
+                    // Path prefix with project name.
+                    var pathPrefix = currentDirectory[..currDirectoryIdx];
+                    // Namespace without project name prefix.
+                    var namespaceSuffix = expectedNamespace[expectedNamespaceIdx..];
 
-                var newPath = Path.Combine(pathPrefix, pathSuffix, fileName);
+                    // `Path.Combine` will return the second argument if it begins with a separation character.
+                    var pathSuffix = namespaceSuffix
+                        .Replace('.', Path.DirectorySeparatorChar)
+                        .Trim(Path.DirectorySeparatorChar);
 
-                if (newPath != document.FilePath)
-                {
-                    var directoryPath = Path.GetDirectoryName(newPath);
+                    var newPath = Path.Combine(pathPrefix, pathSuffix, fileName);
 
-                    if (!Directory.Exists(directoryPath))
+                    if (newPath != document.FilePath)
                     {
-                        Directory.CreateDirectory(directoryPath);
-                    }
+                        var directoryPath = Path.GetDirectoryName(newPath);
+
+                        if (directoryPath != null && !Directory.Exists(directoryPath))
```

Comment:
```suggestion
                        if (directoryPath != null)
```

Plus the whole file could benefit from swapping `== null` and `!= null` to `is null` and `is not null` respectively but I don't exactly remember our current conventions here.